include/envoy/http/access_log.h

```diff
@@ -114,6 +114,16 @@ class RequestInfo {
    * Set whether the request is a health check request or not.
    */
   virtual void healthCheck(bool is_hc) PURE;
+
+  /**
+   * Set the downstream address
+   */
+  virtual void setDownstreamAddress(std::string address) PURE;
```

Comment:
This should be set by the connection manager since it knows the downstream address. There is no reason for the router to set it at all.